### ISSUE TITLE
Doesn't find "ALL" errors

### ISSUE THREAD

**Issue by mrchief** [SublimeLinter/SublimeLinter-jshint#27]
It seems like it doesn't find all types of errors. For example, linting the following code produces no errors:

``` js
exports = module.exports = HTTPTransport;

var HttpResponseParser = require('./http_response_parser'),
  http = require('http'),
  url = require('url'),
  _ = require('underscore');

function HTTPTransport(logger) {
  this._logger = logger;
}

HTTPTransport.prototype.process = function process(reqObj, callback, err, cData) {
  var errResp,
    urlparts,
    requestHeaders,
    httpRequest,
    self;

  httpRequestObj = {
    'hostname': urlparts.hostname,
    'port': urlparts.port,
    'method': reqObj.method,
    'path': urlparts.path,
    'headers': requestHeaders
  };

  httpRequest.on('error', function (err) {
    var errResp = new ErrorResponse("", (err || {})
      .message, self.options.contenttype)
      .response;
    callback(400, {}, errResp);
  });

  httpRequest.end();
};

```

Console output

```
SublimeLinter: javascript enabled (using node.js)
58
error: No lint errors.
```

At the very least, it should complain about undefined variables such as `httpRequestObj` and `ErrorResponse`

**Screenshots**

Sublime linter is too happy!
![image](https://f.cloud.github.com/assets/781818/2102487/6d0fbae0-8f5f-11e3-9444-818ef8415717.png)

JSHint.com not so much!
![image](https://f.cloud.github.com/assets/781818/2102492/7f2a3336-8f5f-11e3-8083-cec80263ae7a.png)
